debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/FindInstances.java

```diff
@@ -0,0 +1,165 @@
+	}
+
+	private boolean parseArgs(PrintStream out, String[] args) throws DDRInteractiveCommandException {
+		if (args.length != 1) {
+			out.append("Invalid number of arguments" + nl);
+			return false;
+		}
+		String firstArg = args[0];
+		if (firstArg.equals("help")) {
+			printUsage(out);
+			return false;
+		}
+		className = firstArg;
+		return true;
+	}
+
+	public void run(String command, String[] args, Context context, PrintStream out) throws DDRInteractiveCommandException
+	{
+		if (args != null) {
+			if (!parseArgs(out, args)) {
+				return;
+			}
+		}
+
+		try {
+			J9JavaVMPointer vm = J9RASHelper.getVM(DataType.getJ9RASPointer());
+			ClassSegmentIterator iterator = new ClassSegmentIterator(vm.classMemorySegments());
+
+			out.println(
+				String.format(
+					"Searching for class named '%1$s' in VM=%2$s",
+					className, Long.toHexString(vm.getAddress())));
+
+			/* Reset classPointer for each invocation of the command. */
+			classPointer = null;
+
+			while (iterator.hasNext()) {
+				J9ClassPointer clsPointer = (J9ClassPointer)iterator.next();
+				String javaName = J9ClassHelper.getJavaName(clsPointer);
+				if (className.equals(javaName)) {
+					String hexString = clsPointer.getHexAddress();
+					classPointer = clsPointer;
+					out.println(String.format("Found !j9class %1$s named %2$s", hexString, javaName));
+					break;
+				}
+			}
+
+			if (classPointer == null) {
+				out.println(String.format("No class named %1$s found", className));
+				return;
+			}
+		} catch (CorruptDataException e) {
+			throw new DDRInteractiveCommandException(e);
+		}
+
+		/* Allocate a new instance of Table for each invocation of the command. */
+		data = new Table("Instances of " + className);
+		if (data == null) {
+			out.append("Failed to allocate an instance of Table" + nl);
+			return;
+		}
+
+		data.row("Address", "Class Name");
+
+		/* Reset counts for each invocation of the command. */
+		corruptCount = 0;
+		objectsFound = 0;
+
+		scanHeap();
+
+		out.println(String.format("Objects found: %1$d", objectsFound));
+		out.println(String.format("Corruptions encountered: %1$d", corruptCount));
+
+		data.render(out);
+	}
+
+	private void scanHeap() {
+		try {
+			GCHeapRegionIterator regions = GCHeapRegionIterator.from();
+			while (regions.hasNext()) {
+				GCHeapRegionDescriptor region = regions.next();
+				scanObjects(region);
+			}
+		} catch (CorruptDataException e) {
+			e.printStackTrace();
+		}
+
+	}
+
+	private void scanObjects(GCHeapRegionDescriptor region) throws CorruptDataException	{
+		GCObjectHeapIterator heapIterator = GCObjectHeapIterator.fromHeapRegionDescriptor(region, true, true);
+		while (heapIterator.hasNext()) {
+			J9ObjectPointer object = heapIterator.next();
+			try {
+				J9ClassPointer objClass = J9ObjectHelper.clazz(object);
+				if (!objClass.isNull() && J9ClassHelper.isSameOrSuperClassOf(classPointer, objClass)) {
```

Comment:
Suggest using `notNull()` instead of `!isNull()`:
```java
				if (objClass.notNull() && J9ClassHelper.isSameOrSuperClassOf(classPointer, objClass)) {
```